src/crypto/OlmDevice.js

```diff
@@ -815,14 +824,23 @@ OlmDevice.prototype.decryptGroupMessage = async function(
             plaintext = res;
         } else {
             // Check if we have seen this message index before to detect replay attacks.
+            // If the event ID and timestamp are specified, and the match the event ID
+            // and timestamp from the last time we used this message index, then we
+            // don't consider it a replay attack.
             const messageIndexKey = senderKey + "|" + sessionId + "|" + res.message_index;
-            if (messageIndexKey in self._inboundGroupSessionMessageIndexes) {
+            if (messageIndexKey in self._inboundGroupSessionMessageIndexes
+                && (argumentsLength <= 4 // Compatibility for older old versions.
+                    || self._inboundGroupSessionMessageIndexes[messageIndexKey].id !== eventId
+                    || self._inboundGroupSessionMessageIndexes[messageIndexKey].timestamp !== timestamp)) {
```

Comment:
to fix the lint, pull out a local var as @krombel suggested in riot-dev. `tmp` is a rubbish name though.

Reply:
I know why I called it "suggestion" [here](https://matrix.to/#/!DdJkzRliezrwpNebLk:matrix.org/$1507805215244Ltngz:msgs.tk) :sunglasses: